### ISSUE TITLE
SNaP 2.0 Beta Testing

### ISSUE THREAD

**Issue by ClassyTeddy** [CCDevNelson/Snap-Updates#302]
Here is some information on the 2.0 Beta such as how to install, what needs testing, what we know isn't ready, and where to post found bugs and issues.

Testing environment setup instructions can be found here: [SNaP-2.0 README](https://github.com/Classy-Closets/SNaP-2.0)

Important Note:
When testing SNaP 2.0 make sure to start with an empty project directory and create new projects and rooms. Blender files from version 2.79 (SNaP 1.2.x) and below **are not compatible** with Blender 2.8 and 2.9 (SNaP 2.0). 

The following is a general list of what is ready for testing in SNaP 2.0:
- Loading SNaP defaults
- Project manager
    - Creating, deleting, opening rooms
    - Project info
- Room builder
    - Single wall
    - L shape room
    - U shape room
    - Square room
    - Custom room
    - Entryways/doorways
    - Molding, paint, carpet
    - Wall height, wall depth
- Drag and drop product placement (Carcasses/Opening Starters)
- Closet material assignment system
    - Updated materials
    - Updated material UV wrapping. Woodgrain materials now map correctly to the part faces. No distortion should be found from the angle of the viewport/camera or from manipulating part geometry through drivers and prompts.
    - Material textures should not stretch or distort when changing sizes
- Opening starters  
    - Opening starter prompts page
    - Simplified dogear prompts
    - Added functionality to dog-ear partitions per opening
- New right-click options menu for closets
- New right-click options menu to move closets
- New right-click and delete closet products
- Closet Options/defaults
- Theme
- Object library - Decor objects
    - Bathroom
    - Closet
    - Kitchen
    - Misc
    - Outlets and switches
- Create empty assemblies  
- Assembly context UI panel
    - Adding objects to assemblies
    - Enter object edit mode from  assembly panel
    - Select vertex groups and connect hooks from the assembly panel
    - Adding, viewing, and setting prompts from the assembly panel
    - View and manipulate driver logic
- Object context UI panel
    - Prompts
    - Drivers
    - Material Pointers
- Assembly creation process through assembly and object context menus

Some of what is ready for testing might not be relevant to some.

The following is a list of known issues and items that are not ready for testing:
- Drag and drop insert placement (Closet Library - Inserts: issue #13) - Many of the inserts use their own custom drop operator or the insert drop operator. The way in which they were placed into openings and connected to prompts/drivers of other assemblies now causes multiple cyclical dependency errors in Blender 2.9.

- Closet carcass backing (Fix Closet Carcass Backing: issue #16) - Same cyclical dependency errors as above.

- XML Export - This depends on inserts being drawn into products/Opening Starters in order to test.

- 2D Views

Please create issues in the 'Bugs' column of the 'Update to version 2.0' project board: [Project - Update to version 2.0](https://github.com/Classy-Closets/SNaP-2.0/projects/2).

If you have any questions about the Beta, please reply to this post.

**Comment by SeanRadvan** [comment]
Made sure im logged into my classy github account, but im 404ing when i
click the read me.

**Comment by ClassyTeddy** [comment]
That means you currently don't have access to the repository. I am making sure that @RyanMDev gives everyone access. I will let you know when that happens.

**Comment by TedCC1209** [comment]
@ClassyTeddy I had some questions about the directions (this will make sense to everyone one else once they have access):

1. Download SNaP-Blender-2.90.1.zip from Release - Version 2.0 Beta.
2. Extract to desired location
3. Create new folder named scripts.
4. Inside scripts folder, create new folder named addons.
5. Clone or move SNaP-2.0 repository into the addons folder.
6. Set 'Scripts' path in preferences to the point to the scripts folder that was just created.
7. Restart SNaP.

For Number 3, where am I putting the new folder named scripts? Just right in the first main folder I just downloaded and extracted?

![SNaP 2 0](https://user-images.githubusercontent.com/61025894/107256234-7625eb80-69f6-11eb-8e5d-068eb44fea4f.png)

Then I am thinking I will need further direction with number 6 as well as to where I am setting that path within preferences. 

Thanks, 

Ted

**Comment by TedCC1209** [comment]
Verifying the following is correct:

![SNaP 2 0](https://user-images.githubusercontent.com/61025894/107257801-4e378780-69f8-11eb-8919-f8611aa235d1.png)

![image](https://user-images.githubusercontent.com/61025894/107257864-5d1e3a00-69f8-11eb-94a0-3914bc8ca657.png)

![image](https://user-images.githubusercontent.com/61025894/107257924-6c9d8300-69f8-11eb-951e-74fcf2780eb7.png)

**Comment by ClassyTeddy** [comment]
> For Number 3, where am I putting the new folder named scripts? Just right in the first main folder I just downloaded and extracted?

You can put the folder named `scripts` anywhere you want. You just need to be able to find it again later in step 6.

> Then I am thinking I will need further direction with number 6 as well as to where I am setting that path within preferences.

So, to set the path for where Blender needs to find the `scripts` folder by opening Blender, clicking on the `Edit` Button found in the top left of the program, opening the `Preferences`, opening the `File Path` tab, then click on the little folder icon next to the `Scripts` box, and then find and select the `scripts` folder that you just created.
![SNaP 2 0 Scripts Example](https://user-images.githubusercontent.com/62624110/107257878-60b1c100-69f8-11eb-9f9f-674ba498c686.png)

The above that you just sent is correct.

Let me know if that needs more clarification.

**Comment by ClassyTeddy** [comment]
> The above that you just sent is correct.

This is actually not correct, I just tried it myself. You need to just put the `snap` folder found inside the SNaP-2.0 repository. It should be one of the first folders that you find when you open the `SNaP-2.0` folder.

**Comment by TedCC1209** [comment]
This is how it looks after I cloned from GitHub--are you saying I need to cut the highlighted snap folder and move it to the addons folder? Will this mess with syncing with GitHub?

![SNaP 2 0](https://user-images.githubusercontent.com/61025894/107259660-8049e900-69fa-11eb-972e-09e597d1e589.png)

**Comment by ClassyTeddy** [comment]
> This is how it looks after I cloned from GitHub--are you saying I need to cut the highlighted snap folder and move it to the addons folder? Will this mess with syncing with GitHub?

Yes you need to do that. As I understand it, it won't be synced with GitHub if you use the testing environment instructions. It will be as if you are an end user who just downloaded the entire program. @RyanMDev, correct me if I am wrong on this.

**Comment by TedCC1209** [comment]
Looks like moving the snap folder to the addons folder rather than being in the SNaP-2.0 folder worked, thanks!

**Comment by TylerStandage** [comment]
Just to be clear Ted, you just moved the 'snap' folder up into the addons
folder like I've done below, correct?

[image: image.png]

**Comment by TedCC1209** [comment]
Yes:

![SNaP 2 0](https://user-images.githubusercontent.com/61025894/107262463-d2d8d480-69fd-11eb-81a5-27dd82254e6f.png)

**Comment by TylerStandage** [comment]
After updating the SNaP defaults, I can no longer see the project manager/room builder/etc. menus.  They were there the first time I opened SNaP, but after the defaults loaded they went away. 

![Capture1](https://user-images.githubusercontent.com/77167318/107271124-a4102d80-6a00-11eb-9f66-07b8c48c18dc.PNG)

**Comment by ClassyTeddy** [comment]
This just occired to me. Go back into preferences and select the scripts file path again like you did in step 6

**Comment by TylerStandage** [comment]
That worked, thanks!

**Comment by RyanMDev** [comment]
> > This is how it looks after I cloned from GitHub--are you saying I need to cut the highlighted snap folder and move it to the addons folder? Will this mess with syncing with GitHub?

It will, after moving the snap folder cloned from GitHub you can then go back into GitHub desktop and relocate the repository. In order for this to work, you will also need to copy over the .git folder with it as well, this is what tells GitHub Desktop that this is a git repository. With this set up correctly, the GitHub repo functionality stays intact, allowing access to the most recent changes.

![reloc_snap](https://user-images.githubusercontent.com/6529544/107283311-15f07300-6a11-11eb-8382-1f0d3299a198.png)
![reloc_snap2](https://user-images.githubusercontent.com/6529544/107283334-1db01780-6a11-11eb-8047-7e5bdd086365.png)

**Comment by sarahhightower** [comment]
Hi. I've cloned the snap folder to the noted location and made sure it's linked to the scripts folder in SNaP. As I look at this though, I'm not seeing any "Classy Closets" functions so I think I missed a step somewhere?  Could it be that I can't find the ".git" folder in Github as noted by Ryan in his last message? 

![1](https://user-images.githubusercontent.com/72047269/107398565-fa42a680-6abc-11eb-8bcd-4b56810ece6c.png)

![2](https://user-images.githubusercontent.com/72047269/107398572-fd3d9700-6abc-11eb-8152-7ec019d41aef.png)

**Comment by ClassyTeddy** [comment]
No, that is not the problem @sarahhightower. Your script path isn't correct. The path for the script folder needs to be `C:\Users\shightower\De...0.1\scripts` not `C:\Users\shightower\De...0.1\scripts\add ons\snap\`.

Also, the addons folder needs to be one word I believe.

**Comment by sarahhightower** [comment]
That worked! Final questions (I hope!).

My SNaP doesn't have the same color scheme as Tyler's. Is that something I should be concerned about or is that something I can just change? Also, @RyanMDev can you give me a bit more detail on copying over the .git file? I'm not sure how you got to the first prompt you show in your message. Thank you.

![1](https://user-images.githubusercontent.com/72047269/107403045-e2215600-6ac1-11eb-8d64-473ac8ab2737.png)

**Comment by sarahhightower** [comment]
Nevermind, @RyanMDev. I think I figured it out. You just made a folder named .git and then copied the Github SNaP 2.0 file in it, correct?

**Comment by ClassyTeddy** [comment]
@sarahhightower In order to get the right color scheme, you need to Load SNaP Defaults, and when you do that, you will have to set the script path again.

**Comment by RyanMDev** [comment]
The `.git` folder is created by the GitHub app when you clone or create a repository. It only affects the GitHub app and not anything in SNaP.

You only need to move this file if you see this message in the GitHub app:
> ![reloc_snap](https://user-images.githubusercontent.com/6529544/107283311-15f07300-6a11-11eb-8382-1f0d3299a198.png)

This would happen if you cloned the repository first and then manually moved it in the file explorer to the new `scripts/addons` folder. If  you do not move the `.git` folder along with it you will not be able to locate and restore it in the GitHub app.

> I've cloned the snap folder to the noted location

It sounds like you have created the `scripts/addons` folder first and then cloned the `SNaP-2.0` repo directly into that folder. In this case, which would be the preferred way to do it, the `.git` folder will already be in place.

This is just to make sure you still have access to sync the newest repo changes in Github.

**Comment by TylerStandage** [comment]
Hi everyone, 

I'm reviving this thread to ensure we all have the right steps for getting 2.0 ready for testing.  I've also added a screen shot to show where my folders are located after following the steps. I've placed my 2.0 on the desktop, but you can place it where makes for you on your computer.  Most important is that you do all of the steps and set the scripts path correctly. The back and forth in this thread is helpful, but let me know if you have questions! 

1. Download SNaP-Blender-2.90.1.zip from Release - Version 2.0 Beta.
2. Extract to desired location.
3. Create new folder named scripts.
4. Inside scripts folder, create new folder named addons.
5. Clone or move SNaP-2.0 repository into the addons folder.
6. Set 'Scripts' path in preferences to the point to the scripts folder that was just created.
7. Restart SNaP.

![image](https://user-images.githubusercontent.com/77167318/112880394-3cc93e00-907f-11eb-90b4-fd083d001adc.png)
